nodefs/loopback_linux_test.go

```diff
@@ -198,3 +202,119 @@ func TestCopyFileRange(t *testing.T) {
 	}
 
 }
+
+// Wait for a change in /proc/self/mounts
+func waitMount() error {
+	fd, err := syscall.Open("/proc/self/mounts", syscall.O_RDONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer syscall.Close(fd)
+	pollFds := []unix.PollFd{
+		{
+			Fd:     int32(fd),
+			Events: unix.POLLPRI,
+		},
+	}
+	_, err = unix.Poll(pollFds, 1000)
+	return err
+}
+
+// There is a hang that appears when enabling CAP_PARALLEL_DIROPS on Linux
+// 4.15.0: https://github.com/hanwen/go-fuse/issues/281
+// The hang was originally triggered by gvfs-udisks2-volume-monitor. This
+// test emulates what gvfs-udisks2-volume-monitor does.
+func TestParallelDiropsHang(t *testing.T) {
+	// We do NOT want to use newTestCase() here because we need access to
+	// mntDir before the filesystem is mounted
+	dir := testutil.TempDir()
+	orig := dir + "/orig"
+	mnt := dir + "/mnt"
+	if err := os.Mkdir(orig, 0755); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Mkdir(mnt, 0755); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// Unblock the goroutines onces the mount shows up in /proc/self/mounts
+	var wait sync.RWMutex
```

Comment:
nit: I suggest to use a channel to trigger start here:

```
start := make(chan struct{})
...
// when mounted
close(start)
...
// users do
<-start
```

Reply:
Yeah that's nicer. Fixed.